### PR TITLE
Make global usage of use in header 

### PR DESCRIPTION
**Pull request by titanrat**
Working on #78.
If someone want to help - you are welcome.

### CODE REVIEW

**Comment by PowerKiKi** [comment]
Thanks for contributing to that. Please take great care to strictly change only the namespace things, as this will be a huge PR to review already. And let me know when you think you are done...

File: docs/topics/recipes.md [hunk 1]
```diff
@@ -743,7 +743,7 @@ easy manner.
 ### Valid array keys for style `applyFromArray()`
 
 The following table lists the valid array keys for
-\PhpOffice\PhpSpreadsheet\Style applyFromArray() classes. If the "Maps
+Style applyFromArray() classes. If the "Maps
```
Comment (gmsantos) [inline docs/topics/recipes.md]: 👎 in documentation like this

Comment (PowerKiKi) [inline docs/topics/recipes.md]: I would tend to agree that fully qualified class names might be better for documentation. That way the user don't need to look for where the class lives.

I guess we could have exceptions to that in specific context where it is rather obvious, but most class names in documentation should probably stay in FQCN form.

Comment (titanrat) [inline docs/topics/recipes.md]: Ok. I'll revert all ".md" files before final commit.

Comment (PowerKiKi) [inline docs/topics/recipes.md]: btw, do you have a rough estimate on when you could finish this work ? I am thinking on releasing a beta "soonish"

File: src/PhpSpreadsheet/Calculation/Statistical.php [hunk 2]
```diff
@@ -3,6 +3,8 @@
 namespace PhpOffice\PhpSpreadsheet\Calculation;
 
 /* LOG_GAMMA_X_MAX_VALUE */
+use PhpOffice\PhpSpreadsheet\Calculation;
+
 define('LOG_GAMMA_X_MAX_VALUE', 2.55e305);
```
Comment (gmsantos) [inline src/PhpSpreadsheet/Calculation/Statistical.php]: namespace importing is between a comment and define. Not sure if this constant should be there (maybe refactor to a class constant)

Comment (PowerKiKi) [inline src/PhpSpreadsheet/Calculation/Statistical.php]: There are indeed a few constants that should be refactored into class constants. But that should be a separate work. I just created #157 for that.

Comment (gmsantos) [inline src/PhpSpreadsheet/Calculation/Statistical.php]: 👍

File: docs/topics/recipes.md (hunk 1 above)
Comment (titanrat) [inline docs/topics/recipes.md]: @PowerKiKi 3-5 days i think.

Comment (PowerKiKi) [inline docs/topics/recipes.md]: Great, there is no rush, but then I'll wait for it for the beta.

**Comment by titanrat** [comment]
@PowerKiKi I think i am done.

File: src/PhpSpreadsheet/Calculation/Statistical.php (hunk 2 above)
Comment (titanrat) [inline src/PhpSpreadsheet/Calculation/Statistical.php]: @gmsantos All "*.md" files reverted.

File: docs/topics/recipes.md (hunk 1 above)
Comment (titanrat) [inline docs/topics/recipes.md]: @gmsantos All ".md" files reverted.

**Comment by PowerKiKi** [comment]
thanks for the work, I'll have a look in the coming days...

File: src/PhpSpreadsheet/Worksheet/ColumnIterator.php [hunk 3]
```diff
@@ -186,10 +183,7 @@ public function next()
     public function prev()
     {
         if ($this->position <= $this->startColumn) {
-            throw new PhpSpreadsheetException('Column is already at the beginning of range (' .
-                Cell::stringFromColumnIndex($this->endColumn) . ' - ' .
-                Cell::stringFromColumnIndex($this->endColumn) . ')'
-            );
+            throw new PhpSpreadsheetException('Column is already at the beginning of range (' . Cell::stringFromColumnIndex($this->endColumn) . ' - ' . Cell::stringFromColumnIndex($this->endColumn) . ')');
```
Comment (gmsantos) [inline src/PhpSpreadsheet/Worksheet/ColumnIterator.php]: imo multiline is easier to read. What about refactor this to use `sprintf` ?

File: src/PhpSpreadsheet/Worksheet/RowCellIterator.php [hunk 4]
```diff
@@ -170,9 +167,7 @@ public function next()
     {
         do {
             ++$this->position;
-        } while (($this->onlyExistingCells) &&
-            (!$this->subject->cellExistsByColumnAndRow($this->position, $this->rowIndex)) &&
-            ($this->position <= $this->endColumn));
+        } while (($this->onlyExistingCells) && (!$this->subject->cellExistsByColumnAndRow($this->position, $this->rowIndex)) && ($this->position <= $this->endColumn));
```
Comment (gmsantos) [inline src/PhpSpreadsheet/Worksheet/RowCellIterator.php]: I prefer the previous form here too.

File: src/PhpSpreadsheet/Worksheet/ColumnIterator.php [hunk 5]
```diff
@@ -9,20 +9,17 @@
...
- *
  * This library is distributed in the hope that it will be useful,
  * but WITHOUT ANY WARRANTY; without even the implied warranty of
  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
  * Lesser General Public License for more details.
- *
  * You should have received a copy of the GNU Lesser General Public
  * License along with this library; if not, write to the Free Software
- * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
```
Comment (gmsantos) [inline src/PhpSpreadsheet/Worksheet/ColumnIterator.php]: @PowerKiKi will beta version be released as MIT license? I remind to read something about that. These license notices are annoying ahaha

Comment (gmsantos) [inline src/PhpSpreadsheet/Worksheet/ColumnIterator.php]: found it #140

Comment (PowerKiKi) [inline src/PhpSpreadsheet/Worksheet/ColumnIterator.php]: yes, it will

File: src/PhpSpreadsheet/Worksheet/ColumnIterator.php (hunk 3 above)
Comment (PowerKiKi) [inline src/PhpSpreadsheet/Worksheet/ColumnIterator.php]: IMHO, code formatting changes are out of scope of this work, and this should ideally be reverted.

Comment (titanrat) [inline src/PhpSpreadsheet/Worksheet/ColumnIterator.php]: @PowerKiKi codeSniffer fails without this change https://travis-ci.org/PHPOffice/PhpSpreadsheet/jobs/228794493

Comment (PowerKiKi) [inline src/PhpSpreadsheet/Worksheet/ColumnIterator.php]: Fair enough, leave it as is then

File: src/PhpSpreadsheet/Worksheet/RowCellIterator.php (hunk 4 above)
Comment (titanrat) [inline src/PhpSpreadsheet/Worksheet/RowCellIterator.php]: @gmsantos CodeSniffer fails without this change like in previous comment.

File: src/PhpSpreadsheet/Worksheet/ColumnIterator.php (hunk 3 above)
Comment (gmsantos) [inline src/PhpSpreadsheet/Worksheet/ColumnIterator.php]: @PowerKiKi its not the case to review this rule?

**Comment by PowerKiKi** [comment]
Thanks a lot for the contribution :+1: 

I squashed and merged with a few edits (some `as` were unnecessary, and some FQCN were incorrectly lowercase).

**Comment by PowerKiKi** [comment]
For shits and giggles I just counted that we saved a whopping 0.76% bytes in `src/` with this PR (from 4680 to 4716 bytes) :tada:
